app/forms/questionnaire_form.py

```diff
@@ -43,9 +43,13 @@ def validate(self):
 
         for question in self.schema.get_questions_for_block(self.block_json):
             if question['type'] == 'DateRange':
-                valid_date_range_form = self.validate_date_range_question(question)
```

Comment:
As a refactor comment (not necessarily to do now) I think this would be much easier to read if the variables were called `invalid` and the relevant logic inverted.

Reply:
Possible refactor option:

```
    def validate(self):
        """
        Validate this form as usual and check for any form-level date-range/calculated validation errors
        :return:
        """
        is_valid = FlaskForm.validate(self)

        validators_by_type = {
            'DateRange': self.validate_date_range_question,
            'Calculated': self.validate_calculated_question
        }

        for question in self.schema.get_questions_for_block(self.block_json):
            validator = validators_by_type.get(question['type'])
            if validator and not validator(question):
                is_valid = False

        return is_valid
```